### PR TITLE
Modules KomaMRICore and KomaMRIPlots

### PR DESCRIPTION
**Pull request by beorostica**
Refactor with the subdirectory KomaMRICore

### CODE REVIEW

File: KomaMRICore/Project.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+JLD2 = "033835bb-8acc-5ee8-8aae-3f567f8a3819"
+LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+MAT = "23992714-dd62-5051-b70f-ba57cb901cac"
+MRIBase = "f7771a9a-6e57-4e71-863b-6e4b6a2f17df"
+MRIFiles = "5a6f062f-bf45-497d-b654-ad17aae2a530"
+Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
+Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
+PlotlyJS = "f0f68f2c-4968-5e81-91da-67840de0976a"
+ProgressMeter = "92933f4c-e287-5a05-a399-4b506db050ca"
+Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
```
Comment (cncastillo) [inline KomaMRICore/Project.toml:23]: Is random being used?

File: KomaMRICore/src/KomaMRICore.jl [hunk 2]
```diff
@@ -0,0 +1,79 @@
+module KomaMRICore
+
+#IMPORT PACKAGES
+import Base.*, Base.+, Base.-, Base./, Base.vcat, Base.size,
+       Base.copy, Base.Threads.@spawn, Base.Threads.@threads,
+       Base.angle, Base.abs, Base.getproperty, Base.one, Base.zero
```
Comment (cncastillo) [inline KomaMRICore/src/KomaMRICore.jl:6]: This section can be cleaned.

File: KomaMRICore/src/KomaMRICore.jl [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+#General
+using Random, Reexport, ThreadsX
+@reexport using Pkg
```
Comment (cncastillo) [inline KomaMRICore/src/KomaMRICore.jl:9]: Do we really need to reexport Pkg?

File: KomaMRICore/src/KomaMRICore.jl [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+#Printing
+using Scanf, ProgressMeter
+#Datatypes
+using Parameters
+#Simulation
+@reexport using CUDA, Interpolations
```
Comment (cncastillo) [inline KomaMRICore/src/KomaMRICore.jl:15]: Do we really need to reexport CUDA and Interpolations?

File: KomaMRICore/src/KomaMRICore.jl [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+#Display
+export plot_seq, plot_grads_moments, plot_kspace, plot_phantom_map, plot_signal, plot_M0, plot_image, plot_dict
+#Simulator
+export simulate, simulate_slice_profile
+
+export print_gpus
+
+#Additionals
+export get_flip_angles, is_RF_on
+using PlotlyJS
```
Comment (cncastillo) [inline KomaMRICore/src/KomaMRICore.jl:77]: Why export only is_RF_on and not for all the others: ADCs, Grads?

File: KomaMRICore/src/sequences/PulseDesigner.jl [hunk 6]
```diff
@@ -4,8 +4,8 @@
 A module to define different pulse sequences.
 """
 module PulseDesigner
-using ..KomaMRI
-using ..KomaMRI: γ, Scanner, get_bvalue, get_max_grad
+using ..KomaMRICore
+using ..KomaMRICore: γ, Scanner, get_bvalue, get_max_grad
```
Comment (cncastillo) [inline KomaMRICore/src/sequences/PulseDesigner.jl:8]: Is this `using ..KomaMRICore: γ, Scanner, get_bvalue, get_max_grad` necessary?

File: KomaMRICore/src/ui/DisplayFunctions.jl [hunk 7]
```diff
@@ -124,13 +124,13 @@ plot_seq(seq::Sequence; width=nothing, height=nothing, slider=true, show_seq_blo
...
 	if show_sim_blocks
 		#This is the preparation of the default simulate function
-		t, _ = KomaMRI.get_uniform_times(seq, 1e-3)
-		breaks = KomaMRI.get_breaks_in_RF_key_points(seq,t)
+		t, _ = KomaMRICore.get_uniform_times(seq, 1e-3)
+		breaks = KomaMRICore.get_breaks_in_RF_key_points(seq,t)
 		Nt = length(t)
 		if Nblocks == 0
 			Nblocks = ceil(Int, 6506*Nt/1.15e6)
 		end
```
Comment (cncastillo) [inline KomaMRICore/src/ui/DisplayFunctions.jl:132]: I think these part is deprecated.

File: src/KomaMRI.jl [hunk 8]
```diff
@@ -1,78 +1,21 @@
...
-@reexport using MRIFiles: ISMRMRDFile
-#IO
-using FileIO, HDF5, MAT, JLD2
+using Reexport
+@reexport using KomaMRICore
+import KomaMRICore: update_blink_window_progress!
+
 #GUI
 using Blink, Interact, PlotlyJS, AssetRegistry
 @reexport using PlotlyJS: savefig
```
Comment (cncastillo) [inline src/KomaMRI.jl:10]: Is this necessary?

Comment (cncastillo) [inline src/KomaMRI.jl:10]: Is this necessary?

File: KomaMRICore/src/KomaMRICore.jl (hunk 2 above)
Comment (beorostica) [reply KomaMRICore/src/KomaMRICore.jl]: Ok! It will be cleaned!

File: KomaMRICore/src/KomaMRICore.jl (hunk 3 above)
Comment (beorostica) [reply KomaMRICore/src/KomaMRICore.jl]: It will be used __VERSION__.

File: KomaMRICore/src/KomaMRICore.jl (hunk 4 above)
Comment (beorostica) [reply KomaMRICore/src/KomaMRICore.jl]: They will not be reexported!

File: KomaMRICore/src/KomaMRICore.jl (hunk 5 above)
Comment (beorostica) [reply KomaMRICore/src/KomaMRICore.jl:74]: There will be exported: get_flip_angles, is_RF_on, is_GR_on, is_ADC_on

File: KomaMRICore/src/sequences/PulseDesigner.jl (hunk 6 above)
Comment (beorostica) [reply KomaMRICore/src/sequences/PulseDesigner.jl]: Apparently Scanner is necessary. γ, get_bvalue and get_max_grad will be removed.

File: KomaMRICore/src/ui/DisplayFunctions.jl (hunk 7 above)
Comment (beorostica) [reply KomaMRICore/src/ui/DisplayFunctions.jl:132]: It is still necessary to avoid division by zero.

File: src/KomaMRI.jl (hunk 8 above)
Comment (beorostica) [reply src/KomaMRI.jl]: It is not necessary to reexport. However savefig() is used directly in many examples of the documentation, so it is convenient to export savefig().

File: KomaMRICore/src/sequences/PulseDesigner.jl (hunk 6 above)
Comment (cncastillo) [reply KomaMRICore/src/sequences/PulseDesigner.jl]: The problem is that we export `Sequence`, `Scanner`, `Grad`, `RF`, `ADC`, `Delay`, `γ`, etc **after** we define the submodule with `include("sequences/PulseDesigner.jl")`. If we put the definition of PulseDesigner in the end with the definition of the UI, it works correctly:

```julia
#Main
export γ #gyro-magnetic ratio [Hz/T]
export Scanner, Sequence, Phantom
export Grad, RF, ADC, Delay, dur

...

#PulseDesigner submodule
include("sequences/PulseDesigner.jl")
export PulseDesigner
#GUI
include("KomaUI.jl")
export KomaUI

...

```

File: KomaMRICore/src/ui/DisplayFunctions.jl (hunk 7 above)
Comment (cncastillo) [reply KomaMRICore/src/ui/DisplayFunctions.jl:132]: I meant this `Nblocks = ceil(Int, 6506*Nt/1.15e6)`, which is no longer the behavior when the number of blocks is not specified.

File: src/KomaMRI.jl (hunk 8 above)
Comment (cncastillo) [reply src/KomaMRI.jl]: I agree, it is fine while it remains in `KomaMRIPlots`.

File: KomaMRICore/src/KomaMRICore.jl (hunk 4 above)
Comment (cncastillo) [reply KomaMRICore/src/KomaMRICore.jl]: We talked about CUDA being used by the default parameters for the UI, which was changed, but I believe we still have the  `@reexport using CUDA`. 

Sidenote: The `@info "Loading GPUs"` and `KomaMRICore.print_gpus()` will be printed with or without a CUDA-capable GPU (before it had an if statement). Not completely sure what the message says on a PC without NVIDIA GPUs, hopefully, "0 CUDA capable device(s).".

**Pull request merged**